Core/MIPS/x86/X64IRCompFPU.cpp

```diff
@@ -153,10 +161,228 @@ void X64JitBackend::CompIR_FCompare(IRInst inst) {
 
 	switch (inst.op) {
 	case IROp::FCmp:
+		switch (inst.dest) {
+		case IRFpCompareMode::False:
+			regs_.SetGPRImm(IRREG_FPCOND, 0);
+			break;
+
+		case IRFpCompareMode::EitherUnordered:
+			regs_.MapWithExtra(inst, { { 'G', IRREG_FPCOND, 1, MIPSMap::NOINIT } });
+			UCOMISS(regs_.FX(inst.src1), regs_.F(inst.src2));
+			// PF = UNORDERED.
+			SETcc(CC_P, R(SCRATCH1));
+			MOVZX(32, 8, regs_.RX(IRREG_FPCOND), R(SCRATCH1));
+			break;
+
+		case IRFpCompareMode::EqualOrdered:
+			regs_.MapWithExtra(inst, { { 'G', IRREG_FPCOND, 1, MIPSMap::NOINIT } });
+			// Clear the upper bits of SCRATCH1 so we can AND later.
+			XOR(32, R(SCRATCH1), R(SCRATCH1));
+			UCOMISS(regs_.FX(inst.src1), regs_.F(inst.src2));
+			// E/ZF = EQUAL or UNORDERED (not exactly what we want.)
+			SETcc(CC_E, R(SCRATCH1));
+			if (regs_.HasLowSubregister(regs_.RX(IRREG_FPCOND))) {
+				// NP/!PF = ORDERED.
+				SETcc(CC_NP, regs_.R(IRREG_FPCOND));
+				AND(32, regs_.R(IRREG_FPCOND), R(SCRATCH1));
+			} else {
+				MOVZX(32, 8, regs_.RX(IRREG_FPCOND), R(SCRATCH1));
+				// Neither of those affected flags, luckily.
+				// NP/!PF = ORDERED.
+				SETcc(CC_NP, R(SCRATCH1));
+				AND(32, regs_.R(IRREG_FPCOND), R(SCRATCH1));
+			}
+			break;
+
+		case IRFpCompareMode::EqualUnordered:
+			regs_.MapWithExtra(inst, { { 'G', IRREG_FPCOND, 1, MIPSMap::NOINIT } });
+			UCOMISS(regs_.FX(inst.src1), regs_.F(inst.src2));
+			// E/ZF = EQUAL or UNORDERED.
+			SETcc(CC_E, R(SCRATCH1));
+			MOVZX(32, 8, regs_.RX(IRREG_FPCOND), R(SCRATCH1));
```

Comment:
here where the comparison sources are fp registers, we could do the recommended xor / cmp / setcc (see other comment) without any complications actually.